de.metas.business/src/main/java/de/metas/inout/impl/InOutDAO.java

```diff
@@ -68,15 +72,35 @@ public I_M_InOut getById(@NonNull final InOutId inoutId)
 		return load(inoutId, I_M_InOut.class);
 	}
 
+	@Nullable
+	@Override
+	public <T extends I_M_InOut> T getById(@NonNull final InOutId inoutId, @NonNull final Class<T> modelClass)
+	{
+		return load(inoutId, modelClass);
+	}
+
 	@Override
 	public I_M_InOutLine getLineById(@NonNull final InOutLineId inoutLineId)
 	{
 		return load(inoutLineId, I_M_InOutLine.class);
 	}
 
+	@NonNull
+	@Override
+	public ImmutableList<InOutId> retrieveByShipperTransportation(@NonNull final ShipperTransportationId shipperTransportationId)
+	{
+		return Services.get(IQueryBL.class)
+				.createQueryBuilder(de.metas.inout.model.I_M_InOut.class)
+				.addEqualsFilter(de.metas.inout.model.I_M_InOut.COLUMNNAME_M_ShipperTransportation, shipperTransportationId)
```

Comment:
```suggestion
				.addOnlyActiveRecordsFilter()
				.addEqualsFilter(de.metas.inout.model.I_M_InOut.COLUMNNAME_M_ShipperTransportation, shipperTransportationId)
```
unless you explicitly want all records (e.g. because you need to delete all referencing records), always filter for active records. It allows us to deactivate and this hide data from metasfresh